### PR TITLE
Improve url parsing

### PR DESCRIPTION
**Pull request by AsafMah**
#### Pull Request Description

Improved url parsing - supports ports and trailing slashes.

---

#### Future Release Comment
_[Add description of your change, to include in the next release]_
_[Delete any or all irrelevant sections, e.g. if your change does not warrant a release comment at all]_

**Breaking Changes:**
- None

**Features:**
- None

**Fixes:**
- Improved url parsing - supports ports and trailing slashes.

### CODE REVIEW

File: data/src/test/java/com/microsoft/azure/kusto/data/ClientTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        tests.put("https://kusto.test.com/", "https://kusto.test.com");
+        tests.put("https://kusto.test.com/test", "https://kusto.test.com");
+        tests.put("https://kusto.test.com:4242", "https://kusto.test.com:4242");
+        tests.put("https://kusto.test.com:4242/", "https://kusto.test.com:4242");
+        tests.put("https://kusto.test.com:4242/test", "https://kusto.test.com:4242");
+        tests.put("https://kusto.test.com;fed=true", "https://kusto.test.com");
+        tests.put("https://kusto.test.com/;fed=true", "https://kusto.test.com");
+        tests.put("https://kusto.test.com/test;fed=true", "https://kusto.test.com");
+        tests.put("https://kusto.test.com:4242;fed=true", "https://kusto.test.com:4242");
+        tests.put("https://kusto.test.com:4242/;fed=true", "https://kusto.test.com:4242");
```
Comment (ohadbitt) [inline data/src/test/java/com/microsoft/azure/kusto/data/ClientTest.java:29]: Please add cases like application insights and aria proxies

**Review by ohadbitt — APPROVED** [review]

File: data/src/main/java/com/microsoft/azure/kusto/data/UriUtils.java [hunk 2]
```diff
@@ -10,14 +11,16 @@ private UriUtils() {
     }
 
     public static String setPathForUri(String uri, String path, boolean ensureTrailingSlash) throws URISyntaxException {
-        if (ensureTrailingSlash && !path.endsWith("/")) {
-            path += "/";
-        }
-        if (!path.startsWith("/")) {
-            path = "/" + path;
+        if (ensureTrailingSlash) {
+            path = StringUtils.appendIfMissing(path, "/");
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/UriUtils.java:15]: What's the purpose of this block? It seems superfluous to me.

Comment (AsafMah) [reply data/src/main/java/com/microsoft/azure/kusto/data/UriUtils.java]: removed

File: data/src/main/java/com/microsoft/azure/kusto/data/UriUtils.java [hunk 3]
```diff
@@ -10,14 +11,13 @@ private UriUtils() {
     }
 
     public static String setPathForUri(String uri, String path, boolean ensureTrailingSlash) throws URISyntaxException {
-        if (ensureTrailingSlash && !path.endsWith("/")) {
-            path += "/";
-        }
-        if (!path.startsWith("/")) {
-            path = "/" + path;
-        }
+        path = StringUtils.prependIfMissing(path, "/");
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/UriUtils.java:14]: @AsafMah Sorry I'm so late with this, but why did you add this line?

The `URIBuilder` will normalize the path, including if it's missing a leading "/"
